### PR TITLE
Various unchecked-uninit improvements

### PR DESCRIPTION
**Pull request by vorner**
* Links to the std for some functions
* Mention the MaybeUninit
* Unify indentation (spaces, not tabs)

I was thinking about providing an example with MaybeUninit. But I'm not sure how much it is usable now, reading through the issues it seems things like `[MaybeUninit::uninit(), 50]` doesn't yet work.

Btw, I feel a bit uneasy about typing the links to the std functions, `mdbook serve` doesn't include the std for me so I can't check for them leading anywhere. Is anything checking them in CI?

### CODE REVIEW

File: src/unchecked-uninit.md [hunk 1]
```diff
@@ -80,6 +80,19 @@ Every control path through that variable's scope must initialize the value
 before it ends, if it has a destructor.
 *[This includes code panicking](unwinding.html)*.
 
+Not being careful about uninitialized memory often leads to bugs and it has been
+decided the [`mem::uninitialized`][uninitialized] function should be deprecated.
+The [`MaybeUninit`] is supposed to replace it as its API wraps many common
```
Comment (Gankro) [inline src/unchecked-uninit.md]:
```suggestion
The [`MaybeUninit`] type is supposed to replace it as its API wraps many common
```

**Comment by vorner** [comment]
> I think this language needs to be stronger, but this is clearly an improvement, so worth landing.

Do you mean of the newly added paragraph, or of the whole chapter?

**Comment by Gankro** [comment]
my understanding is that mem::uninitialized is borderline impossible to use correctly, because we currently pass metadata to llvm on types with invalid states, like `&T`, and `bool`, which basically instantly trigger UB if a value with that type is `mem::uninitialized()` (llvm undef).

**Comment by vorner** [comment]
Do you have an idea when we can expect `MaybeUninit` to get stabilized? If soon enough, it would probably make sense to rewrite the chapter with that one instead of tuning the warnings about `mem::unitialized`.

**Pull request merged**